service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/UploadTranslationsControllerTest.java

```diff
@@ -0,0 +1,209 @@
+        CaseData updatedCaseData = extractCaseData(postMidEvent(
+            CASE_DATA_WITH_ALL_ORDERS.toBuilder()
+                .uploadTranslationsEventData(UploadTranslationsEventData.builder()
+                    .uploadTranslationsRelatedToDocument(RENDERED_DYNAMIC_LIST.toBuilder()
+                        .value(dlElement(UUID_3, "Notice of proceedings (C6)"))
+                        .build())
+                    .build())
+                .build(),
+            "select-document"
+        ));
+
+        assertThat(updatedCaseData.getUploadTranslationsEventData().getUploadTranslationsOriginalDoc()).isEqualTo(
+            DocumentReference.builder()
+                .filename("noticeo_c6.pdf")
+                .build());
+
+    }
+
+    @Test
+    void shouldFinaliseDocumentsAboutToSubmit() {
+        when(documentDownloadService.downloadDocument(TEST_DOCUMENT.getBinaryUrl())).thenReturn(TRANSLATED_DOC_BYTES);
+        when(documentConversionService.convertToPdf(TRANSLATED_DOC_BYTES, TEST_DOCUMENT.getFilename())).thenReturn(
+            CONVERTED_DOC_BYTES);
+        when(documentSealingService.sealDocument(CONVERTED_DOC_BYTES, SealType.BILINGUAL))
+            .thenReturn(SEALED_DOC_BYTES);
+        when(uploadDocumentService.uploadDocument(SEALED_DOC_BYTES,
+            "noticeo_c6-Welsh.pdf",
+            RenderFormat.PDF.getMediaType()))
+            .thenReturn(UPLOADED_TRANSFORMED_DOCUMENT);
+
+        CaseData updatedCaseData = extractCaseData(postAboutToSubmitEvent(
+            CASE_DATA_WITH_ALL_ORDERS.toBuilder()
+                .uploadTranslationsEventData(UploadTranslationsEventData.builder()
+                    .uploadTranslationsRelatedToDocument(RENDERED_DYNAMIC_LIST.toBuilder()
+                        .value(dlElement(UUID_3, "Notice of proceedings (C6)"))
+                        .build())
+                    .uploadTranslationsOriginalDoc(DocumentReference.builder()
+                        .filename("noticeo_c6.pdf")
+                        .build())
+                    .uploadTranslationsTranslatedDoc(TEST_DOCUMENT)
+                    .build())
+                .build()
+        ));
+
+        assertThat(updatedCaseData).isEqualTo(CASE_DATA_WITH_ALL_ORDERS.toBuilder()
+            .noticeOfProceedingsBundle(List.of(element(UUID_3, DocumentBundle.builder()
+                .document(DocumentReference.builder()
+                    .filename("noticeo_c6.pdf")
+                    .build())
+                .translatedDocument(DocumentReference.buildFromDocument(UPLOADED_TRANSFORMED_DOCUMENT))
+                .translationUploadDateTime(now())
+                .build()
+            ), element(UUID_4, DocumentBundle.builder()
+                .document(DocumentReference.builder()
+                    .filename("noticeo_c6a.pdf")
+                    .build())
+                .build()
+            ))).build());
+
+    }
+
+    @Test
+    void shouldNotifyPartiesWhenSubmitted() {
```

Comment:
don't think this test is doing anything other than not failing :D

Reply:
removed